dashboard/src/shared/Auth.ts

```diff
@@ -112,39 +123,43 @@ export class Auth {
   // the k8s api server nowadays defaults to allowing anonymous
   // requests, so that rather than returning a 401, a 403 is returned if
   // RBAC does not allow the anonymous user access.
+  //
+  // Note: This function is only used now by AxiosInstance which
+  // in turn is only used for operators support.
   public static isAnonymous(response: AxiosResponse<any>): boolean {
     const msg = get(response, "data.message") || get(response, "data");
     return typeof msg === "string" && msg.includes("system:anonymous");
   }
 
-  // isAuthenticatedWithCookie() does an anonymous GET request to determine if
+  // isAuthenticatedWithCookie() does a GET request to determine if
   // the request is authenticated with an http-only cookie (there is, by design,
   // no way to determine via client JS whether an http-only cookie is present).
+  //
+  // Note that when using the auth-proxy, anonymous requests never
+  // get to the backend since the auth-proxy requires authentication.
+  // But if this function is incorrectly called when the auth-proxy
+  // is not in use, with a cluster that supports anonymous requests,
+  // it could potentially return a false positive.
   public static async isAuthenticatedWithCookie(cluster: string): Promise<boolean> {
     try {
-      await Axios.get(url.api.k8s.base(cluster) + "/");
+      await this.resourcesClient().CheckNamespaceExists({
+        context: { cluster, namespace: "default" },
+      });
     } catch (e: any) {
-      const response = e.response as AxiosResponse<any>;
       // The only error response which can possibly mean we did authenticate is
       // a 403 from the k8s api server (ie. we got through to k8s api server
       // but RBAC doesn't authorize us).
-      // See note below about anonymous requests.
-      if (response.status !== 403) {
+      if (e.code !== grpc.Code.PermissionDenied) {
         return false;
       }
 
-      // A 403 error response from the auth proxy itself means we did not get
-      // through to the API server but instead were rejected by the auth
-      // proxy (ie. no http-only cookie).
-      // TODO(mnelson): Check why doesn't the auth proxy return a 401 for a request without auth?
-      if (this.is403FromAuthProxy(response)) {
-        return false;
+      // A 403 error response from our APIs server, rather than the
+      // auth proxy, means we are authenticated and did get
+      // through to the API server but were rejected by RBAC.
+      if (this.is403FromAPIsServer(e)) {
+        return true;
       }
-      // Finally, the k8s api server nowadays defaults to allowing anonymous
-      // requests, so that rather than returning a 401, a 403 is returned if
-      // RBAC does not allow the anonymous user access. An http-only cookie
-      // will not result in an anonymous request, so...
-      return !this.isAnonymous(response);
```

Comment:
I think this assumption that I'd made a year ago is actually incorrect, in that the K8s API server will never receive an anonymous request when the auth proxy is being used, since the auth proxy itself redirects to login when unauthenticated. Hence simply removing this condition. It would be required if this function was called when not using the auth proxy, but this is exactly what I fixed (in the LoginForm) in the previous PR, so that it's only used when `config.authProxyEnabled` is true.